src/v/redpanda/admin_server.cc

```diff
@@ -417,15 +483,19 @@ void admin_server::register_raft_routes() {
           auto shard = _shard_table.local().shard_for(group_id);
 
           return _partition_manager.invoke_on(
-            shard, [group_id, target](cluster::partition_manager& pm) mutable {
+            shard,
+            [group_id, target, this, req = std::move(req)](
+              cluster::partition_manager& pm) mutable {
                 auto consensus = pm.consensus_for(group_id);
                 if (!consensus) {
                     throw ss::httpd::not_found_exception();
                 }
                 return consensus->do_transfer_leadership(target).then(
-                  [](std::error_code err) {
-                      throw_on_error(err);
-                      return ss::json::json_return_type(ss::json::json_void());
+                  [this, req = std::move(req)](std::error_code err)
+                    -> ss::future<ss::json::json_return_type> {
+                      co_await throw_on_error(*req, err, model::controller_ntp);
```

Comment:
repeating this comment. it looks incorrect, but maybe the original comment from the first review got lost in the noise:

why when transferring leadership would we want to redirect to the controller leader, and not the leader of the partition to whose leadership is being transferred?

Reply:
Thanks for bumping, this was indeed missing a co_await, fixed.

Reply:
This was one that I had skipped when updating & left the default controller ntp in, because it didn't have the ntp to hand.  But of course it can be fetch from the consensus -- fixed.